app/views/transactions/_details.haml

```diff
@@ -5,10 +5,17 @@
       %h2
         = link_to_unless listing[:deleted], listing[:title], listing_path(id: listing[:id])
 
-      - if tx[:booking]
-        = render partial: "transactions/details/booking", locals: { booking: tx[:booking], price_per_day: tx[:listing_price] }
+      - # if we don't want to show total price or price_per field, we don't want to show this price-break-down at all
+      - unless local_assigns.has_key?(:show_total_price) && show_total_price == false && tx[:listing_price].cents == 0
```

Comment:
I would remove the `local_assigns.has_key?` and instead add the `show_total_price` to all the places where it's needed